### DISCUSSION TITLE
How to use add/mean aggregation in GENConv

### DISCUSSION THREAD

**Discussion by riskiem** [pyg-team/pytorch_geometric#2997] [Q&A] [answered]
I am trying to use GENConv and I am not getting good results with the default softmax. I want to try out add or mean. When I change my code it throws error.

```
class DeeperGCN(torch.nn.Module):
    def __init__(self, hidden_channels, num_layers):
        super(DeeperGCN, self).__init__()

        self.node_encoder = Linear(data.x.size(-1), hidden_channels)
        self.edge_encoder = Linear(data.edge_attr.size(-1), hidden_channels)

        self.layers = torch.nn.ModuleList()
        for i in range(1, num_layers + 1):
         
            conv = GENConv(hidden_channels, hidden_channels, aggr='add',
                            num_layers=2, norm='layer')
            norm = LayerNorm(hidden_channels, elementwise_affine=True)
            act = ReLU(inplace=True)

            layer = DeepGCNLayer(conv, norm, act, block='res+', dropout=0.5,
                                 ckpt_grad=i % 3)
            self.layers.append(layer)

        self.lin = Linear(hidden_channels, 2)

    def forward(self, x, edge_index, edge_attr):
        x = self.node_encoder(x)
        edge_attr = self.edge_encoder(edge_attr)

        x = self.layers[0].conv(x, edge_index, edge_attr)

        for layer in self.layers[1:]:
            x = layer(x, edge_index, edge_attr)

        x = self.layers[0].act(self.layers[0].norm(x))
        

        return self.lin(x)
```

I get the following error:

> ---------------------------------------------------------------------------
> AssertionError                            Traceback (most recent call last)
> <ipython-input-15-4babaf4230f8> in <module>()
> ----> 1 model = DeeperGCN(hidden_channels=64, num_layers=10)
>       2 #optimizer = torch.optim.Adam(model.parameters(), lr=0.1)
>       3 criterion = torch.nn.CrossEntropyLoss()
>       4 #optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
>       5 optimizer = torch.optim.Adam(model.parameters(), lr=0.1)
> 
> > 1 frames
> > /usr/local/lib/python3.7/dist-packages/torch_geometric/nn/conv/gen_conv.py in __init__(self, in_channels, out_channels, aggr, t, learn_t, p, learn_p, msg_norm, learn_msg_scale, norm, num_layers, eps, **kwargs)
> >     101         self.eps = eps
> >     102 
> > --> 103         assert aggr in ['softmax', 'softmax_sg', 'power']
> >     104 
> >     105         channels = [in_channels]
> > 
> > AssertionError: 
> 
> Can anyone help me fix this?

**Comment by rusty1s** [comment, marked as answer]
That's indeed a mistake. I fixed it in master.

**Reply by riskiem** [reply]
Thank you! I downloaded the latest version. I still can't run this code without error.

**Reply by rusty1s** [reply]
This shouldn't be the case. Note that you need to install PyG from master:
```
pip install git+https://github.com/rusty1s/pytorch_geometric.git
```
